lib/pure/parseutils.nim

```diff
@@ -241,6 +241,10 @@ proc parseIdent*(s: string, start = 0): string =
     while i < s.len and s[i] in IdentChars: inc(i)
     result = substr(s, start, i-1)
 
+proc parseChar*(s: string, ident: var char, start = 0): int =
+  ident = s[start]
```

Comment:
check index(not empty)?

Reply:
Done in scanf though.

Reply:
But other parse* function did check the index?

Reply:
I agree with @xflywind. Not that this is any big deal since this proc is much more unlikely - than e.g. parseIdent - to be called independently from scanf. Still it would be better for consistency with the other parseX procs to have the same behaviour. I could do a small PR to fix this since this is merged with this type of content (adding also docstring, a test and maybe updating docstring of parseIdent for consistency): https://play.nim-lang.org/#ix=2Go3

Reply:
done: https://github.com/nim-lang/Nim/pull/16245